### ISSUE TITLE
Facetting on an "array value" breaks boxplots

### ISSUE THREAD

**Issue by gs0-pix4d** [vega/vega-lite#8201]
Please:

- [x] Check for duplicate issues. Please file separate requests as separate issues.
- [x] Describe how to reproduce the bug.
    When using a facet/row on a field that has type array, the boxplots are broken in weird ways. Use the provided specification below and `vl2svg` to reproduce the bug.
- [x] Use the latest versions of Vega and Vega-Lite.
    ```
    ➜  ~ vl2svg --version
    5.2.0
    ➜  ~ vg2svg --version
    5.22.1
    ```
- [x] Provide an [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) spec in JSON:

In the following example, the rendering is broken for Study `["A", "B"]`, Project `2`, using either facet or row: remove the leading `_` from the row specification to try switching between the two (and disable the `facet` accordingly)

```json
{
  "data": {
    "values": [
        {"err": 8,"study": ["A", "B"],"project": "1"},
        {"err": 3,"study": ["A", "B"],"project": "1"},
        {"err": 6,"study": ["A", "B"],"project": "2"},
        {"err": 0,"study":["A", "B"],"project": "2"}
      ]
  },
  "mark": {"type": "boxplot"},
  "encoding": {
    "facet": {"field": "project"},
    "_row": {"field": "project"},
    "y": {"field": "study"},
    "x": {"field": "err","type": "quantitative"}
  }
}
```

Adding the following transform fixes the problem: `"transform": [{"calculate": "join(datum.study)", "as": "study"}],`. In other words, when the field is not an array but a string, everything is alright.

- [x] If applicable, include error messages and screenshots, GIF videos (e.g. using https://www.cockos.com/licecap/), or working examples (e.g. by clicking share in the [Vega-Editor](https://vega.github.io/editor/) or https://bit.ly/vega-lite-blocks)

My tests are run using `vl2svg` cli tool.

Broken rendering:
![image](https://user-images.githubusercontent.com/86949962/171234015-f0de2317-076c-41ab-9151-266cb8c84978.png)

Cast to string:
![image](https://user-images.githubusercontent.com/86949962/171234186-d47abd28-9b37-46d6-a4ea-17fa7244e6d5.png)

A more complex example (using real world data, please excuse the crushed boxes of the plot) looks like that:
![image](https://user-images.githubusercontent.com/86949962/171238937-86647ccd-01e1-41be-b640-ceba57d6d836.png)

You can see that Case 12 seems to work well, and Case 11 has proper data in front of PREFIX-variant4, while all the others are stuck to the top of the plot area.

**Comment by domoritz** [comment]
Can you look the Vega code to see what needs to be fixed and send a pull request?

**Comment by domoritz** [comment]
I don't think we facet per value. If you wanted that, you could explicitly flatten the field first.
